### PR TITLE
Fix unexpected trailing space

### PR DESCRIPTION
**Pull request by blacelle**
This covers https://github.com/cucumber/gherkin-utils/issues/24

It handles not only `Background` but any `stepContainer` (only examples were already properly managing this case).

### CODE REVIEW

File: java/src/main/java/io/cucumber/gherkin/utils/pretty/PrettyHandlers.java [hunk 1]
```diff
@@ -136,6 +136,14 @@ private static String prettyLanguageHeader(String language) {
         return "en".equals(language) ? "" : String.format("# language: %s\n", language);
     }
 
+    private String semiColumnAndName(String name) {
+    	if (name == null || name.isEmpty()) {
```
Comment (mpkorstanje) [inline java/src/main/java/io/cucumber/gherkin/utils/pretty/PrettyHandlers.java:140]: Can name ever be null?

File: java/src/test/java/io/cucumber/gherkin/utils/pretty/PrettyTest.java [hunk 2]
```diff
@@ -416,12 +416,12 @@ void commentsBeforeAndAfterKeywords(){
...
                         "\n" +
                         "  # before Rule\n" +
-                        "  Rule: \n" +
+                        "  Rule:\n" +
                         "  # after Rule\n" +
                         "\n" +
                         "    # after Rule indent\n" +
                         "    # background comment\n" +
-                        "    Background: foo\n" +
+                        "    Background:\n" +
```
Comment (mpkorstanje) [inline java/src/test/java/io/cucumber/gherkin/utils/pretty/PrettyTest.java:424]: Please do create a separate test case for this. That the rule in this test case was nameless was merely a coincidence.

I imagine a feature file with nameless `Feature`, `Rule`, `Background` keywords and possibly `Given`, `When`, `Then` keywords should do.

File: java/src/main/java/io/cucumber/gherkin/utils/pretty/PrettyHandlers.java [hunk 3]
```diff
@@ -172,13 +179,12 @@ private Result appendFeature(
...
                 .append(stepContainer.getKeyword())
-                .append(": ")
-                .append(stepContainer.getName())
+                .append(semiColumnAndName(stepContainer.getName()))
                 .append("\n")
                 .append(prettyDescription(stepContainer.getDescription(), syntax));
     }
 
-    private Result appendRule(
+	private Result appendRule(
```
Comment (mpkorstanje) [inline java/src/main/java/io/cucumber/gherkin/utils/pretty/PrettyHandlers.java:187]: This looks like an unnecessary change.

**Review by mpkorstanje — CHANGES_REQUESTED** [review]
Cheers. A few comments to clean up.

File: java/src/main/java/io/cucumber/gherkin/utils/pretty/PrettyHandlers.java (hunk 1 above)
Comment (blacelle) [reply java/src/main/java/io/cucumber/gherkin/utils/pretty/PrettyHandlers.java:140]: I preferred staying on the safe-side, and followed https://github.com/cucumber/gherkin-utils/blob/main/java/src/main/java/io/cucumber/gherkin/utils/pretty/PrettyHandlers.java#L219

File: java/src/main/java/io/cucumber/gherkin/utils/pretty/PrettyHandlers.java (hunk 3 above)
Comment (blacelle) [reply java/src/main/java/io/cucumber/gherkin/utils/pretty/PrettyHandlers.java:187]: wrong indentation from me

File: java/src/test/java/io/cucumber/gherkin/utils/pretty/PrettyTest.java (hunk 2 above)
Comment (blacelle) [reply java/src/test/java/io/cucumber/gherkin/utils/pretty/PrettyTest.java]: Separate case added.

Blank `Given`, `When` and `Then` are not stripped. It looks like a different situation to me. Do you confirm these should be cleaned? (I expect not as some matching regex may break).

Comment (mpkorstanje) [reply java/src/test/java/io/cucumber/gherkin/utils/pretty/PrettyTest.java]: Nah. Let's leave them out for now. Not the most important case to fix.

**Review by mpkorstanje — COMMENTED** [review]
LGTM, cheers. 

Leaving this open so I've got a marker to make a release. Possibly thursday pending availability.

@xeger if you happen to make a release before me, please take this one along too.

**Review by mpkorstanje — APPROVED** [review]

**Comment by mpkorstanje** [comment]
Released in v8.0.5
